### ISSUE TITLE
.mp3 support in scipy.io

### ISSUE THREAD

**Issue by MechCoder** [scipy/scipy#3536]
Hi. Thanks for the awesome package.
Scipy.io has support to read/write files of .wav format. Are there any plans to include methods to read files of .mp3 and other formats?
Thanks again.

**Comment by raxod502** [comment]
According to [Wikipedia](https://en.wikipedia.org/wiki/MP3), "the MP3 technology became patent-free in the United States on 16 April 2017 when U.S. Patent 6,009,399, held by and administered by Technicolor, expired". Does this change the situation?

**Comment by Eric89GXL** [comment]
I suspect so, as I've read a little bit about Linux distros possibly picking it up now. But someone would have to do the proper legwork to verify it. Might be worth shooting an email to scipy-dev, where you're likely to get a fuller discussion.

**Comment by raxod502** [comment]
@Eric89GXL I'd like to do that, but first I want to make sure nobody has brought this up before. Is there any way to search the scipy-dev archives? For whatever reason, I can't get to a search function from the [info page](https://mail.scipy.org/pipermail/scipy-dev/).

**Comment by MechCoder** [comment]
You can see the post by a 3-year younger me over here (https://mail.scipy.org/pipermail/scipy-dev/2014-April/019791.html)

**Comment by raxod502** [comment]
Yeah, I actually did find that, although only because I knew what month and year to look in.

**Comment by veecos2** [comment]
Was there any update to this ?

**Comment by mdhaber** [comment]
The patents did indeed expire:
https://www.iis.fraunhofer.de/en/ff/amm/consumer-electronics/mp3.html

It sounds like Fedora includes mp3 support fresh out of the box:
https://fedoraproject.org/wiki/Multimedia/MP3

Ubuntu's legal team is more hesitant:
https://discourse.ubuntu.com/t/mp3-patents-expired-include-full-mp3-support-out-of-the-box-like-fedora-does/3694

NumFOCUS might be able to help us figure out whether it's OK. I'll let you know when I hear back from them.

If we find that we can include it:
[minimp3](https://github.com/lieff/minimp3) is CC0-licensed and appears to be actively maintained.

**Comment by mdhaber** [comment]
NumFOCUS' legal team responded:
"As far as I am aware, the general technology needed to read and write mp3s is not currently restricted in any jurisdiction. I don't think there would be any legal reason not to include support."

**Comment by cchadha2** [comment]
If we're good to go on this and it's not actively being worked on, I wouldn't mind giving it a go?

**Comment by mdhaber** [comment]
I think that sounds great! What package are you thinking of using, and how are you thinking of wrapping it?

**Comment by cchadha2** [comment]
I was thinking of investigating the [minimp3](https://github.com/lieff/minimp3) project you've linked above - for which there are existing [Python bindings](https://github.com/pyminimp3/pyminimp3). The latter is not actively maintained but could be a nice place to start for me to figure out how to create our own bindings here? Or would you suggest contributing there directly?

**Comment by mdhaber** [comment]
There seems to be enough interest here, or there has been in the past. Maybe test out the existing bindings and get a better sense of their status. That would help us decide where your work should go (here or there). 

Are you able to test on multiple platforms? If not, if you have test files and code you've written for one platform, I can try to run it on others.

**Comment by cchadha2** [comment]
That sounds good to me! I can test on Linux and Windows 10 64 bit anyway. May need to lean on you for others.

**Comment by cchadha2** [comment]
@mdhaber Looks like the code at [pyminimp3](https://github.com/pyminimp3/pyminimp3) is not in a working state. Tests (sporadically) pass due to the case where no data is read from the mp3 file, not being tested. I also run into random `ValueError: Out of data` exceptions in the tests which occur for different tests in each run.

Outside of the tests simply using the `load_frame` function creates an object of type `array('h')` which is empty - see [here](https://github.com/pyminimp3/pyminimp3/issues/10#issuecomment-592044245) for more details.

In my opinion, next steps would be to figure out how to create our own bindings for minimp3 here instead of piecing apart and fixing the code over there?

**Comment by mdhaber** [comment]
ok, making our own is fine. Are you able to figure out how it works in C? If so, I can probably help you wrap it with Cython. What functions do we need? Can you put together some simple C code that loads a file into an array?

**Comment by dschmitz89** [comment]
Given popular packages such as [librosa](https://librosa.org/doc/latest/advanced.html), I am closing this as unplanned.
